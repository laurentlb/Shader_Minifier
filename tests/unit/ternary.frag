#version 450

// https://github.com/laurentlb/Shader_Minifier/issues/320
float f0() {
    float a = 1., b = 1.;
    a == 1. ? b = 2. : b = 3.;
    return b; // 2
}

float f1() {
    float a = 1., b = 1.;
    a = true ? b = 2. : b = 3.;
    return a;
}
