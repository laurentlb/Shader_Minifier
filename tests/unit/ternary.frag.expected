#version 450

float f0()
{
  return 2.;
}
float f1()
{
  float b=1.;
  return b=2.;
}
